config/websocket.js

```diff
@@ -0,0 +1,82 @@
+
+import jwt from 'jsonwebtoken';
+import config from './config';
+import notificationCtrl from '../server/controllers/notification.controller';
+
+let server;
+let sockets = [];
+
+function setServer(io) {
+  server = io;
+  server.on('connection', (socket) => {
+    socketEvents(socket);
+  });
+}
+
+function askRegister(socket) {
+  socket.emit('app.register');
+}
+
+function registerSocket(socket, token) {
+  try {
+    const data = jwt.verify(token, config.jwtSecret);
+    if (!data.username) return false;
+    socket.userData = data; // eslint-disable-line no-param-reassign
+    socket.registered = true; // eslint-disable-line no-param-reassign
+    socket.join('registered');
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+function socketEvents(socket) {
+  sockets.push(socket);
+  askRegister(socket);
+
+  socket.on('disconnect', () => {
+    clearDisconnected();
+  });
+
+  socket.on('register', ({ token }) => {
+    const register = registerSocket(socket, token);
+    if (register) socket.emit('app.registered');
+
+    notificationCtrl.sendNotifications(socket.userData._id);
+  });
+
+  socket.on('markread.all', () => {
+    if (socket.userData && socket.userData._id) {
+      notificationCtrl.markReadAll(socket.userData._id);
+    }
+  });
+}
+
+function clearDisconnected() { sockets = sockets.filter(s => s.connected); }
```

Comment:
Your code looks good overall. My only comment is that, with lines like this, I think they should be formatted like below to improve readability and diffability — so if it changes in the future, it's clear what was changed.

```
function clearDisconnected() {
  sockets = sockets.filter(s => s.connected);
}
```

Reply:
Agreed!